### PR TITLE
Image promotion for cluster-api-azure v1.10.4

### PR DESCRIPTION
**Pull request by nojnhuh**
Image promotion for cluster-api-azure v1.10.4
This is an automated PR generated from `kpromo`
```
kpromo pr --fork nojnhuh --project cluster-api-azure --reviewers "@CecileRobertMichon @jackfrancis @mboersma @nojnhuh @sonasingh46" --tag v1.10.4
```

/hold
cc: @CecileRobertMichon @jackfrancis @mboersma @nojnhuh @sonasingh46

### CODE REVIEW

File: registry.k8s.io/images/k8s-staging-cluster-api-azure/images.yaml [hunk 1]
```diff
@@ -35,6 +35,7 @@
     "sha256:66e0c46112a03c0239c3feed0607ae85aa068d75dd6db73c938070be13724658": ["v1.10.1"]
     "sha256:7e132ae90d334cecb4c6fff2eda16ffcc63136c47e69f843400e1df4820c81ff": ["v1.10.2"]
     "sha256:b551551e7035b56313a29f001deca50afca7c122912d068c344930688f24e66e": ["v1.10.3"]
+    "sha256:0dbdcc88313b75be28d6fdf31c905d18034709823032cecdd0a1963c81dcd469": ["v1.10.4"]
```
Comment (nojnhuh) [inline registry.k8s.io/images/k8s-staging-cluster-api-azure/images.yaml:38]: Looks good :+1: 

https://console.cloud.google.com/gcr/images/k8s-staging-cluster-api-azure/global/cluster-api-azure-controller@sha256:0dbdcc88313b75be28d6fdf31c905d18034709823032cecdd0a1963c81dcd469/details

**Review by mboersma — COMMENTED** [review]
/lgtm

**Pull request merged**